plugins/Morpheus/javascripts/piwikHelper.js

```diff
@@ -23,6 +23,42 @@ function _pk_translate(translationStringId, values) {
     return "The string "+translationStringId+" was not loaded in javascript. Make sure it is added in the Translate.getClientSideTranslationKeys hook.";
 }
 
+function _pk_externalRawLink(url, values) {
+
+  if (!url) {
+    return '';
+  }
+
+  const campaignOverride = (typeof values != 'undefined' && values.length > 0 && values[0] ? values[0] : null);
+  const sourceOverride = (typeof values != 'undefined' && values.length > 1 && values[1] ? values[1] : null);
+  const mediumOverride = (typeof values != 'undefined' && values.length > 2 && values[2] ? values[2] : null);
+
+  const returnURL = new URL(url);
+  const validDomain = returnURL.host.endsWith('matomo.org');
+  const urlParams = new URLSearchParams(window.location.search);
+  const module = urlParams.get('module');
+  const action = urlParams.get('action');
+
+  // Apply campaign parameters if domain is ok, config is not disabled and a value for medium exists
+  if (validDomain && !window.piwik.disableTrackingMatomoAppLinks
+    && ((module && action) || mediumOverride)) {
+    const campaign = (campaignOverride === null ? 'Matomo_App' : campaignOverride);
+    let source = (window.Cloud === undefined ? 'OnPremise' : 'Cloud');
```

Comment:
The PHP version has these as 'Matomo_App_OnPremise' and 'Matomo_App_Cloud', should they be the same here or adjust in PHP so it's consistent?

Reply:
Good spot, I've updated the JS version to be consistent with PHP :+1: